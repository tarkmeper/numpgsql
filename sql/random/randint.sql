SELECT random_seed(1);

SELECT random_randint(1,5,'{2}'::integer[]);

SELECT random_randint(1,5,'{2,2}'::integer[]);

-- some error-ish conditions
SELECT random_randint(5,1,'{2,2}'::integer[]);
SELECT random_randint(1,1,'{2,2}'::integer[]);
