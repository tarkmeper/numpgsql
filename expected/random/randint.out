SELECT random_seed(1);
 random_seed 
-------------
 
(1 row)

SELECT random_randint(1,5,'{2}'::integer[]);
 random_randint 
----------------
 {3,5}
(1 row)

SELECT random_randint(1,5,'{2,2}'::integer[]);
 random_randint 
----------------
 {{4,5},{1,1}}
(1 row)

-- some error-ish conditions
SELECT random_randint(5,1,'{2,2}'::integer[]);
ERROR:  GEN_RANDINT: Cannot have high bound lower then low bound
SELECT random_randint(1,1,'{2,2}'::integer[]);
ERROR:  GEN_RANDINT: Cannot have high bound lower then low bound
